doc/pages/expansions.asciidoc

```diff
@@ -1,199 +1,300 @@
+Kakoune's `\*debug*` buffer. If you're trying to debug a shell expansion,
+check the debug buffer with `:buffer \*debug*` to see if anything shows up.
 
-*kak_selections*::
-    quoted list of the contents of the selections.
+Because Kakoune does not expand expansions inside the text of an expansion,
+you can't use normal expansions inside `%sh{}`. Instead, Kakoune can export
+expansions as environment variables to make them available to the shell.
+Here's how expansion patterns map to variable names:
 
-*kak_selection_desc*::
-    range of the main selection, represented as anchor,cursor; anchor
-    and cursor are in this format: line.column
+*%arg{n}*::
+    (where _n_ is a decimal number) +
+    becomes `$_n_`. For example, `%arg{3}` becomes `$3`.
 
-*kak_selections_desc*::
-    unquoted list range of the selections.
+*%arg{@}*::
+    becomes `$@`
 
-*kak_bufname*::
-    name of the current buffer
+*%opt{x}*::
+    becomes `$kak_opt_x`
 
-*kak_buffile*::
-    full path of the file or same as kak_bufname when there’s no
-    associated file
+*%reg{x}*::
+    (where _x_ is the alphabetic name of a register) +
+    `$kak_reg_x` contains all the selections in register _x_ +
+    `$kak_main_reg_x` contains only the main selection
 
-*kak_buflist*::
-    quoted list of the currently opened buffer names
+*%val{x}*::
+    becomes `$kak_x`
 
-*kak_buf_line_count*::
-    the current buffer line count
+When turned into environment variables, list-type options, `$kak_reg_x`, and
+"quoted list" values will be shell-quoted so the shell doesn't get confused
+about how many items the list contains. You will need to apply `eval` to get
+back the original values. For example, if you want to process the contents
+of each selection, you can do something like:
 
-*kak_timestamp*::
-    timestamp of the current buffer, the timestamp is an integer value
-    which is incremented each time the buffer is modified
+----
+eval set -- $kak_selections
+while [ $# -gt 0 ]; do
+    # ...do a thing with $1...
+    shift
+done
+----
 
-*kak_history_id*::
-    history id of the current buffer, the history id is an integer value
-    which is used to reference a specific buffer version in the undo tree
+The `eval` command will take the expanded `$kak_selections` and unquote them,
+then execute the resulting `set` command, which sets the shell's argument
+variables to the items from `$kak_selections`. The `while` loop with `shift`
+iterates through the arguments one by one.
 
-*kak_runtime*::
-    directory containing the kak support files, determined from kakoune's
-    binary location.
+Only variables actually mentioned in the body of the shell expansion will
+be exported into the shell's environment. For example:
 
-*kak_config*::
-    directory containing the user configuration
+----
+echo %sh{ env | grep ^kak_ }
+----
 
-*kak_version*::
-    version of the current Kakoune server (git hash or release name)
+...will find none of Kakoune's special environment variables, but:
```

Comment:
I am surprised by `...` not being followed by a space, but it might well be the usual way to write it, is that correct ?

Reply:
My personal habit is that an ellipsis gets space on one side or the other depending on whether it's trailing off ...

... or resuming a previous thought. However, now that I check Wikipedia, it has [any number of contradictory guidelines](https://en.wikipedia.org/wiki/Ellipsis#In_American_English), none of which really cover this specific circumstance. 

I'll add a space.